pkg/ingester/rate_calculator.go

```diff
@@ -34,6 +37,10 @@ func (c *RateCalculator) recordSample() {
 	for range t.C {
```

Comment:
This works but a more idiomatic golang pattern is to use a stop chan of type `struct{}`

```go
for {
    select {
    case <-c.stopChan:
        return
    case <-t.C:
        rate := c.sample.Swap(0)
        c.storeNewRate(rate)		
    }
}
```

Then `Stop` can just
```go
close(c.stopChan)
```

Reply:
I pushed a commit using a channel for it instead, but since I'm a newb with channels, I'm not sure the way I did it is idiomatic. WDYT?

Reply:
I'd put the `t.C` case on the top because it's the thing I want someone to see first when they read this part, but that's stylistic. `select` statements are not evaluated in order so there's no behavioral difference.

lgtm